### ISSUE TITLE
Manually closing collapse

### ISSUE THREAD

**Issue by Aspie96** [thednp/bootstrap.native#64]
Hi.

Is there any way to manually close a collapse created dynamically through:
```javascript
var menuCollapse = new Collapse("#menu-button");
```

There is a `.close(e)` method, but it requires an event parameter.

Is there any way I can do it manually, asynchronously? It'd be really useful.
Thank you very much!

**Comment by thednp** [comment]
Thanks for spotting that out, I will look into it and provide a solution asap.

**Comment by thednp** [comment]
I just improved this, but it doesn't work properly in IE8, currently working on a fix.

**Comment by Aspie96** [comment]
How so?

The event is added through:
```javascript
this.btn.removeEventListener('click', this.toggle, false);
```

Therefore, if I am getting this right (please, correct me if not), the trigger should always be `self.btn` (and you can easily get the trigger from the target).

**Comment by thednp** [comment]
Indeed you are wrong, the collapse instance does not know which one is `self.btn` until `self.toggle` is executed. I am going to try to find a way to simplify the script while keeping same visual and browser support.

**Comment by Aspie96** [comment]
Mhh, I don't quite get it, honestly.

When initializing each collapse, `this.btn` gets set to `element` (which is the button) and the event handler `this.toggle` gets first bind to such element.
`toggle` then sets it to `self.getTarget(e).btn`.
`getTarget` sets the return value `btn` to `e.currentTarget`. `e.currentElement` is at this point the original button because it is what has been clicked (since any other click would not have invoked the event).

So, whileas it's true that `this.btn` gets reset, it will actually be set to the same value each time.

I made a few tries with any collapse you have in the sapmple page (the menu one, the accordion and the one with two buttons), saving a pointer to the original button in a new field (`myButton`), and it is the case.
What I did was adding `this.myButton = this.btn` after the very first initialization of this.btn and then `console.log(self.btn === self.globalButton);` after every call of `self.getTarget` (only after which `self.button` might be modified) and it printed all trues.

I even removed every reintialization of `self.btn` and it works just fine.

**Comment by thednp** [comment]
I have a new version of the file, un-minified only at this time. Would you like to test it in your app?

**Comment by Aspie96** [comment]
I just downloaded the last version of native-accordion.js and repeated the two tests, getting the exact same results.

**Comment by thednp** [comment]
Wait, I haven't committed yet.

**Comment by Aspie96** [comment]
Ah, ok, I apologize.

Anyways, there was no need to modify it for this purpose as, as I said, "self.btn" never changes.

**Comment by thednp** [comment]
OK now, go ahead and download, test and let me know.

**Comment by Aspie96** [comment]
It does not work managing it manually.
I haven't checked why yet, but I would guess it is because of self.collapse not being set.

As self.btn never seems to change, nor the target does.
Therefore, this.collapse can easily be initialized in the Collapse constructor and getTarget is not actually needed.

**Comment by thednp** [comment]
I was thinking the same while working on the code tonight, when I originally developed the script I was thinking of a solid solution for the use of accordions. I am going to try to change it now, and let you know.

**Comment by Aspie96** [comment]
But it does work with accordion as well.

You create a brand new instance of Collapse for each single button (even if the accordion is the exact same).
So you have no reason at all to worry about accordions.

**Comment by Aspie96** [comment]
You literally cycle throw all buttons (not collapsible items themselves) creating instances:
```javascript
var Collapses = document.querySelectorAll('[data-toggle="collapse"]'), i = 0, cll = Collapses.length;
for (i;i<cll;i++) {
	var item = Collapses[i], options = {};
	options.duration = item.getAttribute('data-duration');
	new Collapse(item,options);
}
```

**Comment by thednp** [comment]
The 'getTarget' method is needed in any case, we only know the btn as the element, we need to find out the target collapse. So there is a new commit, please download and try again.

**Comment by Aspie96** [comment]
> The 'getTarget' method is needed in any case, we only know the btn as the element, we need to find out the target collapse. So there is a new commit, please download and try again.
Yeah, yeah, of course.
I meant that it would only be executed once (so it could not be a method on its own, if one wanted to remove it, but it can stay there for simplicity).
But it does not need to change the button, at all, nor it ever needs a event handler.

**Comment by thednp** [comment]
The new code doesn't change the btn and collapse on every `open()` or `close()` execution now, only once on `init()`. Please check in your app.

**Comment by Aspie96** [comment]
Anyways, it is late here in Italy (01:10), so I am gonna quit thinking about JavaScript now, until tomorrow.

See you soon and thank you very much for your work.

If it helps you, I can modify the code myself, but not right now.
Just tell me if you want me to and I sure will.

Thank you again. See you tomorrow.

**Comment by thednp** [comment]
Please test tomorrow the new code if you can do whatever you wanted in the first place.

**Comment by Aspie96** [comment]
I sure will, thank you again.
Not now, since I couldn't even think rationally at 1:14. Not a good time for debugging, at least for me.

**Comment by Aspie96** [comment]
Ok, I just tested it and it seems to work well and nicely.

Thank you very much again.

**Comment by thednp** [comment]
Thanks for your input, I appreciate it.

**Comment by RyanZim** [comment]
@thednp Can this be released soon?

**Comment by thednp** [comment]
I believe so.

**Comment by RyanZim** [comment]
> Perhaps you can go through your own tests and spot your own little thing

Not sure what you mean.

**Comment by RyanZim** [comment]
:grinning: It seems that's how it goes. I'll do a little work on the latest master and report back.

**Comment by thednp** [comment]
Well, I think you are too late, I am gonna release 1.0.3 in a few minutes.

**Comment by thednp** [comment]
Ok now, you can play around with the latest release.

**Comment by RyanZim** [comment]
@thednp `npm publish` please?

**Comment by thednp** [comment]
I don't do that myself, I had a friend that did that, if you can contribute with that, I am happy.

**Comment by RyanZim** [comment]
@thednp It's not hard.

I assume you have NodeJS installed.

Open the terminal and navigate to the bootstrap.native directory. (Make sure git is at the commit that you tagged as v1.0.3). Type `npm publish` and hit enter!

If that throws an error, you will need to run `npm login`. This will prompt you for your email, username, and password for npmjs.com. (If you forgot your password, you can recover it at https://www.npmjs.com/forgot). Then run `npm publish` again.

I'm not familiar with bower, perhaps @IngwiePhoenix can help.

**Comment by thednp** [comment]
I think I did it :dancer:

**Comment by RyanZim** [comment]
:tada: You did!

**Comment by thednp** [comment]
My first ever deploy to the npm.
